Shared/Samples/Search for web map/SearchForWebMapView.swift

```diff
@@ -0,0 +1,189 @@
+                            SafeMapView(map: Map(item: item))
+                                .navigationTitle(item.title)
+                        } label: {
+                            PortalItemRowView(item: item)
+                        }
+                        .buttonStyle(.plain)
+                        .onAppear {
+                            // Load the next results when the last item is reached.
+                            if item.id == model.portalItems.last?.id {
+                                model.findNextItems()
+                            }
+                        }
+                    }
+                }
+                
+                ProgressView()
+                    .padding()
+                    .opacity(resultsAreLoading ? 1 : 0)
+            }
+            
+            VStack {
+                Text("No Results")
+                    .font(.headline)
+                Text("Check spelling or try a new search.")
+                    .font(.footnote)
+            }
+            .opacity(!resultsAreLoading && !query.isEmpty && model.portalItems.isEmpty ? 1 : 0)
+        }
+        .background(Color(.secondarySystemBackground))
+        .searchable(
+            text: $query,
+            placement: .navigationBarDrawer(displayMode: .always),
+            prompt: "Search for a web map"
+        )
+        .onChange(of: query) { _ in
+            // Load new results when the query changes.
+            model.findItems(for: query)
+        }
+        .task(id: model.task) {
+            guard model.task != nil else { return }
+            
+            resultsAreLoading = true
+            defer { resultsAreLoading = false }
+            
+            do {
+                try await model.task?.value
+            } catch {
+                self.error = error
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension SearchForWebMapView {
+    /// A map view that shows an alert and dismisses itself when there is an error loading the map.
+    struct SafeMapView: View {
+        /// The map to show in the map view.
+        let map: Map
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// A Boolean value indicating whether the map is being loaded.
+        @State private var mapIsLoading = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        
+        var body: some View {
+            ZStack {
+                MapView(map: map)
+                    .task {
+                        mapIsLoading = true
+                        defer { mapIsLoading = false }
+                        
+                        // Show an alert for an error loading the map.
+                        do {
+                            try await map.load()
+                        } catch {
+                            self.error = error
```

Comment:
I'm discussing the error handling with Nimesh - there are lots of items on public AGOL where they use an Esri basemap, and some private layers, all in the same item. It causes  authentication problems that says

```
ArcGIS Maps SDK
Invalid API Key Error
The API key '{XXX}' is invalid, expired, or does not have access to the resource '{XXX}'.
Please visit https://developers.arcgis.com/api-keys to configure the API key to access the resource, or to create a new API key.
```

Ideally we can capture those errors as well, in addition to map load error.